db/migrate/20160419134542_add_deployment_authentication_options_to_authentications.rb

```diff
@@ -0,0 +1,35 @@
+class AddDeploymentAuthenticationOptionsToAuthentications < ActiveRecord::Migration[5.0]
+  def change
+    add_column :authentications, :auth_challenge, :boolean
+    add_column :authentications, :auth_login, :boolean
+    # from old table t.string :kind assuming we could use auth_type is the correct? (could be ldap/htpass etc..)
```

Comment:
Yes, you can use a combination of STI (so, say, `AuthenticationLdap < Authentication`) and authtype which is an arbitrary string that is interpreted by the caller, so it could be `default` or `backup` or whatever.